src/backend/joanie/core/views.py

```diff
@@ -46,84 +59,57 @@ class DebugMailSuccessPaymentViewTxt(DebugMailSuccessPayment):
     template_name = "mail/text/order_validated.txt"
 
 
-class DebugCertificateDefinitionTemplateView(TemplateView):
+class DebugTemplateView(TemplateView):
     """
-    Debug View to check the layout of certificate definition templates
+    Simple class to render the PDF template in bytes format of a document to preview.
+    Issuer document can be a : certificate, degree, invoice, or a contract definition.
     """
 
-    type = None
+    model = None
+    issuer_document = None
+    template_name = "debug/pdf_viewer.html"
 
-    # pylint: disable=invalid-name, unused-argument
-    def retrieve_certificate_from_pk(self, pk, site):
+    def get_document_context(self):
         """
-        Retrieve a certificate from its pk and its certificate definition template
-        and generate its document
+        Base method to get document's context for a given document.
         """
-        certificate = Certificate.objects.get(
-            pk=pk,
-            certificate_definition__template=self.type,
-        )
-
-        document = issuers.generate_document(
-            name=certificate.definition.template,
-            context=certificate.get_document_context(),
-        )
-
-        return document
+        raise NotImplementedError("subclasses must implement this method.")
 
-    # pylint: disable=unused-argument
-    def generate_certificate(self, site):
+    def retrieve_document_from_pk(self, pk):
         """
-        Generate an enrollment certificate document with hardcoded organization,
-        course, enrollment and certificate definition to prevent to create a lot of
-        objects in the database.
+        Retrieve the object from its 'pk', and generate the context and create
+        the document in PDF bytes in return.
         """
-        organization = factories.OrganizationFactory(
-            code=f"DEV_{self.type.upper()}_ORGANIZATION",
-        )
-
-        course = factories.CourseFactory(
-            code=f"DEV_{self.type.upper()}_COURSE",
-            organizations=[organization],
-        )
-
-        enrollment = factories.EnrollmentFactory(
-            course_run__course=course,
-        )
-
-        definition = factories.CertificateDefinitionFactory.create(
-            name=f"DEV_{self.type.upper()}_DEFINITION",
-            template=self.type,
-        )
-
-        certificate = factories.EnrollmentCertificateFactory.create(
-            certificate_definition=definition,
-            organization=organization,
-            enrollment=enrollment,
+        try:
+            object_to_render = self.model.objects.get(pk=pk)
+        except self.model.DoesNotExist as exception:
+            logger.error(
+                "No object with %s exists for the model %s : %s",
+                pk,
+                self.model,
+                exception,
+            )
+            return None
```

Comment:
As this is for dev purpose, I think it's useful to let's Django display the error so I'll do not use try except here.